syntax = "proto2";

package NCloud.NFileStore.NProto;

option go_package = "a.yandex-team.ru/cloud/filestore/config";

////////////////////////////////////////////////////////////////////////////////

message TClientConfig
{
    optional string Host = 1;
    optional uint32 Port = 2;

    optional uint32 RetryTimeout = 3;
    optional uint32 RetryTimeoutIncrement = 4;

    optional uint32 ThreadsCount = 5;

    // Request timeout (in milliseconds).
    optional uint32 RequestTimeout = 6;

    // Max retry timeout in case of connection failure (in milliseconds).
    optional uint32 ConnectionErrorMaxRetryTimeout = 7;
}

////////////////////////////////////////////////////////////////////////////////

message TClientAppConfig
{
    optional TClientConfig ClientConfig = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TSessionConfig
{
    optional string FileSystemId = 1;
    optional string ClientId = 2;

    optional uint32 SessionPingTimeout = 3;
    optional uint32 SessionRetryTimeout = 4;
}
